doc/source/release_notes/index.rst

```diff
@@ -0,0 +1,14 @@
+Release notes
+=============
+
+This is the list of changes to scikit-image between each release. For full details, see
+the `commit logs`_.
+
+.. _commit logs: https://github.com/scikit-image/scikit-image/commits/
+
+.. naturalsortedtoctree::
+    :maxdepth: 1
+    :glob:
+    :reversed:
+
+    release_[0-9d]*
```

Comment:
@stefanv, you asked to include the release notes in an automated manner. I've done so with a custom directive that is able to sort entries by their natural order which the default `toctree` directive can't do.

Another option is to just include the latest release notes as we did before.

I'm undecided, whether this is the best approach. I could also just add this functionality to `build_modref.py`. I'm slightly in favor of moving away from having to run a custom script before sphinx. Extending sphinx directly with templates and extensions, might be more flexible and simpler to implement.

Though, I think that should mostly come in other PRs. Let me know if I should keep it this way for now or integrate into `build_modref.py`.